src/core/EdFi.DataManagementService.Core/Middleware/ParseBodyMiddleware.cs

```diff
@@ -0,0 +1,41 @@
+// SPDX-License-Identifier: Apache-2.0
+// Licensed to the Ed-Fi Alliance under one or more agreements.
+// The Ed-Fi Alliance licenses this file to you under the Apache License, Version 2.0.
+// See the LICENSE and NOTICES files in the project root for more information.
+
+using System.Text.Json.Nodes;
+using EdFi.DataManagementService.Core.Pipeline;
+using EdFi.DataManagementService.Core.Response;
+using Microsoft.Extensions.Logging;
+
+namespace EdFi.DataManagementService.Core.Middleware
+{
+    internal class ParseBodyMiddleware(ILogger _logger) : IPipelineStep
+    {
+        public async Task Execute(PipelineContext context, Func<Task> next)
+        {
+            _logger.LogDebug("Entering ParseBodyMiddleware - {TraceId}", context.FrontendRequest.TraceId);
+
+            if (context.FrontendRequest.Body != null)
+            {
+                try
+                {
+                    var body = JsonNode.Parse(context.FrontendRequest.Body);
+
+                    if (body != null)
+                    {
+                        context.ParsedBody = body;
+                    }
+                }
+                catch (Exception ex)
+                {
+                    _logger.LogError(ex, "Unknown Error - {TraceId}", context.FrontendRequest.TraceId);
```

Comment:
Its true that this is an unknown error, but we can do a bit better. Imagine reading the log and seeing "Unknown Error" followed by the details. You have to get the stack trace to figure out where this occurred. Perhaps write `"Unable to parse the request body as JSON - {TraceId}"`.

Actually, since this is a 400 response and a client error, it should not be logged _as an error_ on the server. This should just be a DEBUG message.

Reply:
> Its true that this is an unknown error, but we can do a bit better. Imagine reading the log and seeing "Unknown Error" followed by the details. You have to get the stack trace to figure out where this occurred. Perhaps write `"Unable to parse the request body as JSON - {TraceId}"`.
> 
> Actually, since this is a 400 response and a client error, it should not be logged _as an error_ on the server. This should just be a DEBUG message.

Thanks @bradbanister , it was my first solution to the problem, however I thought that this change should be made in many places and that's why I added it in `ValidateEqualityConstraintMiddleware`